ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_math.py

```diff
@@ -2773,3 +2799,41 @@ def test_jax_numpy_polymul(
         atol=1e-05,
         rtol=1e-03,
     )
+
+
+@handle_frontend_test(
+    fn_tree="jax.numpy.product",
+    dtype_x_axis_dtype_where=_get_castable_dtypes_values(use_where=True),
+    keepdims=st.booleans(),
+    initial=st.one_of(st.floats(min_value=-100, max_value=100))
+)
+def test_jax_numpy_product(
+        dtype_x_axis_dtype_where,
+        keepdims,
+        initial,
+        frontend,
+        test_flags,
+        fn_tree,
+        on_device
+):
+    input_dtypes, x, axis, dtype, where = dtype_x_axis_dtype_where
+    if ivy.current_backend_str() == "torch":
+        assume(not test_flags.as_variable[0])
```

Comment:
It is generally not recommended to use `assume` inside the test function itself unless absolutely necessary. Why have we included this check here especially for the `torch` backend?

Reply:
The current `torch` backend does not seem to support this. The numpy.prod test seem to also exclude `torch` backend.

Reply:
Should I leave it as a failing check instead?

Reply:
Right. Yeah sure no worries then. Thanks!